server/zclient.go

```diff
@@ -654,41 +669,27 @@ func (z *zebraClient) loop() {
 						"Topic": "Zebra",
 					}).Debug("MultiplePathListLen %d, PathListLen %d", len(msg.MultiPathList), len(msg.PathList))
 					for _, dst := range msg.MultiPathList {
-						if len(dst) == 0 {
-							continue
-						}
-						path := dst[0]
+						// We need to check all of Path for self route withdraw.
 						selfRouteWithdraw := false
-						if NlriPrefix(path.GetNlri().String()) == "0.0.0.0/0" {
-							continue
-						}
-						if path.IsLocal() {
-							fmt.Println("Make Local Path selection to withdraw event", path.GetNlri().String())
-							selfRouteWithdraw = true
+						for _, path := range dst {
+							if path.IsLocal() {
+								fmt.Println("Make Local Path selection to withdraw event", path.GetNlri().String())
+								selfRouteWithdraw = true
+							}
 						}
-						var vrf uint16 = 0
 						fmt.Println("dst-len", len(dst), "msg.Vrf", msg.Vrf)
-						for index, p := range dst {
+						for pos, path := range dst {
+							if NlriPrefix(path.GetNlri().String()) == "0.0.0.0/0" {
+								continue
+							}
+							var vrf uint16 = 0
```

Comment:
VRF table ID in the kernel is uint32. Is this OK?

Reply:
upstream gobgp uses uint16.  maybe we should consider to expand to uint32 in near future.